### PR TITLE
Update CKEditor 5 descriptions to reflect the fact that now CKEditor 5 is built using TS not ES6

### PR DESCRIPTION
**Pull request by AnnaTomanek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Update CKEditor 5 descriptions to mention that CKEditor 5 is built using TypeScript. Closes cksource/ckeditor5-internal#3036.

### CODE REVIEW

**Comment by AnnaTomanek** [comment]
Also added this to the [Framework overview page](https://ckeditor.com/docs/ckeditor5/latest/framework/index.html).

File: README.md [hunk 1]
```diff
@@ -9,34 +9,38 @@ CKEditor 5 [![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?styl
...
-   * [Other CKEditor 5 installation methods](#ckeditor-5-advanced-installation)
-* [Documentation and FAQ](#documentation-and-faq)
-* [Releases](#releases)
-* [CKEditor 5 features](#editing-and-collaboration-features)
-* [Contributing and project organization](#contributing-and-project-organization)
-   * [Ideas and discussions](#ideas-and-discussions)
-   * [Development](#development)
-   * [Reporting issues and feature requests](#reporting-issues-and-feature-requests)
-* [License](#license)
+- [CKEditor 5 ](#ckeditor-5-)
```
Comment (godai78) [inline README.md:18]: Do we need these two first items in here? It's basically self linking?

**Review by apadol-cksource — APPROVED** [review]
LGTM

**Review by godai78 — APPROVED** [review]
LGTMN

**Pull request merged**